src/functions-and-arrays.js

```diff
@@ -1,119 +1,316 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
-
-
-
+function maxOfTwoNumbers(num1, num2) {
+  if (num1 > num2) {
+    return num1;
+  } else {
+    return num2;
+  }
+}
+console.log(maxOfTwoNumbers(2, 3));
 // Iteration #2: Find longest word
-const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
-
-function findLongestWord() {}
-
-
+const words = [
+  "mystery",
+  "brother",
+  "aviator",
+  "crocodile",
+  "pearl",
+  "orchard",
+  "crackpot",
+];
 
+function findLongestWord(arrays) {
+  let longestWord = arrays[0];
+  let length = arrays.length;
+  if (length === 0) {
+    longestWord = null;
+    return longestWord;
+  } else {
+    for (let i = 0; i < length; i++) {
+      if (arrays[i].length > longestWord.length) {
+        longestWord = arrays[i];
+      }
+    }
+
+    return longestWord;
+  }
+}
+console.log(findLongestWord(words));
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
-
-
+function sumNumbers(arrNumbers) {
+  let sum = 0;
+  let length = arrNumbers.length;
+  if (length === 0) {
+    sum = 0;
+    return sum;
+  }
+  for (let i = 0; i < length; i++) {
+    sum += arrNumbers[i];
+  }
+  return sum;
+}
+console.log(sumNumbers(numbers));
 
 // Iteration #3.1 Bonus:
-function sum() {}
-
+function sum(mixedArr) {
+  let result = 0;
+  for (let i = 0; i < mixedArr.length; i++) {
+    if (typeof mixedArr[i] === "string") {
+      result += mixedArr[i].length;
+    } else if (mixedArr[i] === true) {
+      result += 1;
+    } else if (mixedArr[i] === false) {
+      result += 0;
+    } else if (typeof mixedArr[i] === "number") {
+      result += mixedArr[i];
+    }
+    //else if (Array.isArray(mixedArr[i]) || typeof mixedArr[i] === "object") {
+    //   throw new Error("Unsupported data type");
```

Comment:
The string need to be exactly that one `throw new Error("Unsupported data type sir or ma'am")`